ropetest/contrib/autoimport/deptest.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Iterable
+
+import pytest
+
+from rope.base.project import Project
+from rope.contrib.autoimport.sqlite import AutoImport
+
+
+@pytest.fixture
+def project(request, tmp_path: Path) -> Iterable[Project]:
```

Comment:
There's already a `project` fixture in `ropetest/conftest.py` with a slightly different definition. I'd prefer to avoid having multiple overlapping definitions of fixtures without good reason here.

If we really just want to create pyproject.toml by default, then perhaps make an autouse fixture.

Reply:
Hmm - this one has to write the pyproject before the project is created (since that is when the preferences are parsed). If the project fixture used tmp_path, this would be doable, but the directory isn't created until the fixture in this case. Should I rename this fixture to pyproject?

Reply:
We can make the `project` depends on `pyproject_toml` fixture. The default implementation of `pyproject_toml` could just be a no-op (i.e. no `pyproject.toml`), but tests that wants to have one can override the fixture.

Reply:
Done in 0a94b920